meinberlin/apps/contrib/forms.py

```diff
@@ -0,0 +1,197 @@
+        """
+        Returns the form class to be used with the named form.
+        """
+        return self._get_from_name(name, 'form_class')
+
+    def get_initial(self, name):
+        """
+        Returns the initial data to use for the named form.
+        """
+        initial = self._get_from_name(name, 'initial', {})
+        return initial.copy()
+
+    def get_prefix(self, name):
+        """
+        Returns the prefix to use for the named form.
+        """
+        if self.prefix:
+            return '{}_{}'.format(self.prefix, name)
+        return name
+
+    def get_form_kwargs(self, name):
+        """
+        Returns the keyword arguments for instantiating the named form.
+        """
+        kwargs = {
+            'initial': self.get_initial(name),
+            'prefix': self.get_prefix(name),
+        }
+        if self.request.method in ('POST', 'PUT'):
+            kwargs.update({
+                'data': self.request.POST,
+                'files': self.request.FILES,
+            })
+        return kwargs
+
+    def get_success_url(self):
+        """
+        Returns the supplied success URL.
+        """
+        if self.success_url:
+            # Forcing possible reverse_lazy evaluation
+            url = force_text(self.success_url)
+        else:
+            raise ImproperlyConfigured(
+                "No URL to redirect to. Provide a success_url.")
+        return url
+
+
+class MultiModelFormMixin(MultiFormMixin):
+    objects = {}
+
+    def forms_valid(self, forms):
+        """
+        If all forms are valid, save the associated models.
+        """
+        self.objects = self.forms_save(forms)
+        return super().forms_valid(forms)
+
+    def forms_save(self, forms, commit=True):
+        """
+        Save all the forms in one transaction.
+        """
+        objects = {}
+        with transaction.atomic():
+            for name in self.forms.keys():
+                if hasattr(forms[name], 'save'):
+                    objects[name] = forms[name].save(commit)
+        return objects
+
+    def get_form_class(self, name):
+        """
+        Returns the form class to be used with the named form.
+        """
+        fields = self._get_from_name(name, 'fields')
+        form_class = self._get_from_name(name, 'form_class')
+        model = self._get_from_name(name, 'model')
+
+        if fields is not None and form_class:
+            raise ImproperlyConfigured(
+                "Specifying both 'fields' and 'form_class' is not permitted."
+            )
+
+        if form_class:
+            return form_class
+        elif model is not None:
+            if fields is None:
+                raise ImproperlyConfigured(
+                    "Using MultiModelFormMixin (base class of %s) without "
+                    "the 'fields' attribute is prohibited."
+                    % self.__class__.__name__
+                )
+            return model_forms.modelform_factory(model, fields=fields)
```

Comment:
I think this function is a bit hard to read. Maybe there is a clearer way to distinguish the allowed setting combinations.

Reply:
This is analogous to the core django get_form_class implementation. I would keep it that way for uniformity.